packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart

```diff
@@ -203,12 +203,15 @@ abstract class ImagePickerPlatform extends PlatformInterface {
   /// image types such as JPEG. If compression is not supported for the image that is picked,
   /// a warning message will be logged.
   ///
+  /// The `limit` argument defines how many images can be select.
+  ///
   /// If no images were picked, the return value is null.
   @Deprecated('Use getMultiImageWithOptions instead.')
```

Comment:
This method is deprecated and shouldn't be updated for this change.

The actual implementation of this deprecation is backwards. This method should be calling the newer method rather than the reverse.

Updating that would make this change to the arguments unneeded.

Reply:
Changing this would also make the changes to non ios/android sub-packages unneeded and revertable.

Reply:
as well as clean up some unneeded test changes.

Reply:
I think this is a breaking change, that's why there's the "WithOptions" version, so we can add a new parameter to the constructor of the object (but not to a method that is going to be overridden)

Reply:
@tarrinneal I've created this issue:

* https://github.com/flutter/flutter/issues/144286

Reply:
I have removed the limit from deprecated methods